### PR TITLE
Add initial CI config

### PR DESCRIPTION
**Pull request by anttih**
Adds a CI configuration that runs the test suite using the npm package `purescm`.

### CODE REVIEW

**Comment by anttih** [comment]
@f-f any idea what's up with the build?

**Comment by f-f** [comment]
@anttih the CI is not happy because it's trying to feed to the compiler JS modules that are not ES modules, and the compiler rejects those. I believe this is because we have based this implementation on the previous major version of the library (5.x), and we should rebase all of our commits on the latest commit from upstream.

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -11,22 +11,20 @@ jobs:
...
 
       - name: Install dependencies
         run: |
-          npm install -g bower
-          npm install
-          bower install --production
+          sudo apt install chezscheme
+
+      - name: Setup PureScript dependencies
+        run: npm i --global purescript@0.15.10 purs-tidy@latest spago@next purescript-psa@latest purescm@next
```
Comment (f-f) [inline .github/workflows/ci.yml]: Spago includes PSA now:

```suggestion
        run: npm i --global purescript@0.15.10 purs-tidy@latest spago@next purescm@next
```

File: package.json [hunk 2]
```diff
@@ -1,14 +1,10 @@
 {
   "private": true,
   "scripts": {
-    "clean": "rimraf output && rimraf .pulp-cache",
-    "build": "eslint src && pulp build -- --censor-lib --strict",
-    "test": "pulp test"
+    "build": "spago build",
+    "test": "spago build && purescm bundle-app --main Test.Main && scheme --program output/main"
```
Comment (f-f) [inline package.json]: Just a thought: we might want to add a `test` command to `purescm`, that does the bundle-app + call to scheme

File: package.json [hunk 3]
```diff
@@ -1,14 +1,10 @@
...
   },
   "devDependencies": {
-    "eslint": "^7.15.0",
-    "purescript-psa": "^0.8.0",
-    "pulp": "^15.0.0",
-    "rimraf": "^3.0.2"
+    "purescm": "1.0.0"
```
Comment (f-f) [inline package.json]: I don't think we need this line if we are relying on a global install

**Review by f-f — APPROVED** [review]

**Pull request merged**